ui-v2/app/utils/acls-status.js

```diff
@@ -0,0 +1,48 @@
+// This is used by all acl routes to check whether
+// acls are enabled on the server, and whether the user
+// has a valid token
+// Right now this is very acl specific, but is likely to be
+// made a bit more less specific
+export default function(P = Promise) {
+  return function(obj) {
+    const propName = Object.keys(obj)[0];
+    const p = obj[propName];
+    let authorize;
+    let enable;
+    return {
+      isAuthorized: new P(function(resolve) {
+        authorize = function(bool) {
+          resolve(bool);
+        };
+      }),
+      isEnabled: new P(function(resolve) {
+        enable = function(bool) {
+          resolve(bool);
+        };
+      }),
+      [propName]: p
+        .catch(function(e) {
```

Comment:
Took me a minute to follow these through, but this is a clever way of wrapping the call.